lua/nvim-possession/init.lua

```diff
@@ -171,11 +157,39 @@ M.setup = function(user_opts)
 		}
 		opts = require("fzf-lua.config").normalize_opts(opts, {})
 		opts = require("fzf-lua.core").set_header(opts, { "actions" })
-		fzf.fzf_exec(list_sessions, opts)
+
+		---autoload mechanism
+		if cwd then
+			local sessions_in_cwd = utils.list_sessions(user_config, cwd)
+			if next(sessions_in_cwd) == nil then
+				vim.notify("no session to autoload", vim.log.levels.WARN, { title = notification_title })
+			elseif #sessions_in_cwd == 1 then
+				vim.cmd.source(user_config.sessions.sessions_path .. sessions_in_cwd[1])
+				vim.g[user_config.sessions.sessions_variable] = vim.fs.basename(sessions_in_cwd[1])
+				if type(user_config.post_hook) == "function" then
+					user_config.post_hook()
+				end
+			else
+				fzf.fzf_exec(function(fzf_cb)
+					for _, sess in ipairs(utils.list_sessions(user_config, cwd)) do
+						fzf_cb(sess)
+					end
+					fzf_cb()
+				end, opts)
+			end
+		---standard list load mechanism
+		else
+			fzf.fzf_exec(function(fzf_cb)
+				for _, sess in ipairs(utils.list_sessions(user_config)) do
+					fzf_cb(sess)
+				end
+				fzf_cb()
+			end, opts)
+		end
```

Comment:
nit: to reduce on code duplication, could just do:
```lua
fzf.fzf_exec(function(fzf_cb)
  for _, sess in ipairs(utils.list_sessions(user_config, cwd)) do
    fzf_cb(sess)
  end
  fzf_cb()
end, opts)
```

And then early return on the other cases.

Reply:
I have refactored this in c6a438736a757e0fda146c19c2cda8c241e63806 with early returns, let's check if all cases work as intended!